base/ca/src/org/dogtagpki/ct/CTEngine.java

```diff
@@ -0,0 +1,745 @@
+            while (e.hasMoreElements()) {
+                String n = e.nextElement();
+                Extension ext = (Extension) exts.get(n);
+
+                logger.debug(" ---- " + ext.getExtensionId().toString());
+            }
+        } catch (Exception e) {
+            logger.debug(method + e.toString());
+        }
+        logger.debug(method + "ends");
+    }
+
+    /**
+     * (Certificate Transparency)
+     *
+     * timeStampHexStringToByteArray converts timestamp hex string to bytes
+     *   - example timestamp: 00000172.71270909
+     * @param timeStampString hex string (example above)
+     * @return byte[] timestamp in byte array
+     */
+    public static byte[] timeStampHexStringToByteArray(String timeStampString) {
+        String method = "timeStampHexStringToByteArray: ";
+        int len = timeStampString.length();
+        logger.debug(method + "len =" + len);
+        byte[] data = new byte[(len-1) / 2];
+        for (int i = 0; i < len; i += 2) {
+            if (i == 8 ) {
+                i--; // skip the '.' and at i+=2 it will move to next digit
+                continue;
+            }
+            data[i / 2] = (byte) ((Character.digit(timeStampString.charAt(i), 16) << 4)
+                             + Character.digit(timeStampString.charAt(i+1), 16));
+        }
+        return data;
+    }
+
+    /**
+     * createSCTextension - creates the SCT extension which encompasses
+     * SCT returned from each CT log
+     *
+     * (Certificate Transparency)
+     *
+       https://tools.ietf.org/html/rfc6962
+       ...
+          a certificate authority MAY submit a Precertificate to
+          more than one log, and all obtained SCTs can be directly embedded in
+          the final certificate, by encoding the SignedCertificateTimestampList
+          structure as an ASN.1 OCTET STRING and inserting the resulting data
+          in the TBSCertificate as an X.509v3 certificate extension (OID
+          1.3.6.1.4.1.11129.2.4.2).  Upon receiving the certificate, clients
+          can reconstruct the original TBSCertificate to verify the SCT
+          signature.
+       ...
+
+       SCT response:
+
+       struct {
+           Version sct_version;
+           LogID id;
+           uint64 timestamp;
+           CtExtensions extensions;
+           digitally-signed struct {
+               Version sct_version;
+               SignatureType signature_type = certificate_timestamp;
+               uint64 timestamp;
+               LogEntryType entry_type;
+               select(entry_type) {
+                   case x509_entry: ASN.1Cert;
+                   case precert_entry: PreCert;
+               } signed_entry;
+              CtExtensions extensions;
+           };
+       } SignedCertificateTimestamp;
+
+    * @param ctResponses: list of CTResponse as input param
+    * @return SCT extension
+    */
+    Extension createSCTextension(List<String> ctResponses) {
```

Comment:
Minor: This method needs to be moved into `org.dogtagpki.ct.sct` since this is specific to SCT

Reply:
I"m not sure. I'll think about it.